docs/start/framework/react/learn-the-basics.md

```diff
@@ -275,26 +275,62 @@ Here's a quick example of how you can use server functions to perform a mutation
 
 ```tsx
 import { createServerFn } from '@tanstack/react-start'
+import { z } from 'zod'
+import { db } from '...'
 
 const UserSchema = z.object({
   id: z.string(),
   name: z.string(),
 })
+export type User = z.infer<typeof UserSchema>
 
-const updateUser = createServerFn({ method: 'POST' })
+export const updateUser = createServerFn({ method: 'POST' })
   .validator(UserSchema)
   .handler(async ({ data }) => {
-    return db
-      .update(users)
-      .set({ name: data.name })
-      .where(eq(users.id, data.id))
+    const result = await db.users.update({
+      where: {
+        id: data.id,
+      },
+      data: {
+        name: data.name,
+      },
+    })
   })
 
 // Somewhere else in your application
-await updateUser({ data: { id: '1', name: 'John' } })
+import { useQueryClient } from '@tanstack/react-query'
+import { useRouter } from '@tanstack/react-router'
+import { useServerFunction } from '@tanstack/react-start'
+import { updateUser, type User } from '...'
+
+export function useUpdateUser() {
+  const router = useRouter()
+  const queryClient = useQueryClient()
+  const _updateUser = useServerFunction(updateUser)
+
+  return useCallback(
+    async (user: User) => {
+      const result = await _updateUser({ data: user })
+
+      router.invalidate()
+      queryClient.invalidateQueries({
+        queryKey: ['users', 'updateUser', user.id],
+      })
+
+      return result
+    },
+    [router, queryClient, _updateUser],
+  )
+}
+
+// Somewhere else in your application
+import { useUpdateUser } from '...'
+
+const updateUser = useUpdateUser()
```

Comment:
This assumes its inside the confines of the React component. Please show it as such.